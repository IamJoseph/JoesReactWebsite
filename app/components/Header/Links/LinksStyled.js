import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

export const Container = styled.nav`
  height: 68px;
  margin-right: 1.87em;
  z-index: 1;
  @media(min-width: 1024px){
    margin-right: 3.2em;
  }
`;

 /*
  * CollapsibleMenu
  */

export const Menu = styled.div`
  @media(min-width: 700px){
    display: none;
  }
`;

export const MenuLink = styled(NavLink)`
  background-color: #6d8da7;
  color: #fffef6;
  padding: 1em;
  text-decoration: none;
  z-index: 1;
`;

export const UL = styled.ul`
  border-left: 1px solid #627d93;
  border-right: 1px solid #627d93;
  border-bottom: 2px solid #627d93;
  border-radius: 0 0 3px 3px;
  display: flex;
  flex-direction: column;
  font-size: 1.25em;
  margin-top: 0;
  opacity: 1;
  padding-left: 0;
  position: relative;
  transform: none;
  transition: opacity .1s ease-out .15s, transform .3s linear, z-index 0s linear .4s;
  will-change: transform;
  z-index: 1;
  &.hide{
    opacity: 0;
    transition: opacity .15s ease-in, transform .3s linear, z-index linear;
    transform: translateY(-12em);
    z-index: -9;
  }
`;

export const Button = styled.div`
  align-items: center;
  background-color: #6d8da7;
  border-bottom: 2px solid #627d93;
  display: flex;
  height: 70px;
  justify-content: center;
  width: 120px;
`;

export const IMG = styled.img`
  display: flex;
  height: 30px;
  margin: auto;
`;

 /*
  * Links
  */

export const Link = styled(NavLink)`
  align-items: center;
  align-self: center;
  border-radius: 10px 10px 0 0;
  color: #fffef6;
  cursor: pointer;
  display: none;
  font-size: 1.4em;
  font-weight: 300;
  height: 100%;
  margin-top: 2px;
  outline: 0;
  padding: 0 .7em;
  user-select: none;
  text-decoration: none;
  &.active{
    background-color: #fffef6;
    border-bottom: 2px solid #e3e2de;
    border-left: 1px solid #627d93;
    border-right: 2px solid #627d93;
    border-top: 1px solid #627d93;
    color: #5f5c5a;
  }
  @media(min-width: 700px){
    display: inline-flex;
  }
`;
