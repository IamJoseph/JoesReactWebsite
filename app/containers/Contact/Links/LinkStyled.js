import styled from 'styled-components';

export const Container = styled.div`
display: flex;
margin-top: 105px;
justify-content: center;
`;

export const A = styled.a`
  margin: 0 1em;
`;

export const IMG = styled.img`
  width: 4em;
  @media(min-width: 768px){
    width: 3.125em;
  }
`;
