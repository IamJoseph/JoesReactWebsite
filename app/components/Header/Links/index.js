import React, { Component } from 'react';
import Helmet from 'react-helmet';
import { Container, Menu, MenuLink, UL, IMG, Button, Link } from './LinksStyled';
import menuIcon from '../../../assets/menu_icon.svg';

const activeDropdown = {
  backgroundColor: '#8aa3b6',
};

export default class Links extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };
  }
  /* eslint-disable react/no-did-update-set-state */
  componentDidUpdate(prevProps, prevState) {
    if (prevState.open) {
      this.setState({ open: false });
    }
  }

  clickHandler = () => {
    this.setState({ open: true });
  }

  render() {
    return (
      <Container>
        <Helmet>
          <script type="application/ld+json">
            {`
              {
                  "@context": "http://schema.org",
                  "@type": "Organization",
                  "name": "Joseph Ireland",
                  "url": "https://www.joeireland.com",
                  "logo": "https://www.joeireland.com/d695e9d11a028511bb06553bfca19888.svg"
              }
            `}
          </script>
        </Helmet>
        <Menu>
          <Button>
            <IMG onClick={this.clickHandler} src={menuIcon} alt="hamburger menu icon" />
          </Button>
          <UL className={!this.state.open && 'hide'}>
            <MenuLink exact activeStyle={activeDropdown} to="/">Home</MenuLink>
            <MenuLink activeStyle={activeDropdown} to="/work">Work</MenuLink>
            <MenuLink activeStyle={activeDropdown} to="/contact">Contact</MenuLink>
          </UL>
        </Menu>
        <Link exact activeClassName="active" to="/">Home</Link>
        <Link activeClassName="active" to="/work">Work</Link>
        <Link activeClassName="active" to="/contact">Contact</Link>
      </Container>
    );
  }
}
