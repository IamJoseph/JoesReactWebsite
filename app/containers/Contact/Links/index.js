import React from 'react';
import facebookIcon from '../../../assets/facebook_logo.svg';
import githubIcon from '../../../assets/github_logo.svg';
import linkedInIcon from '../../../assets/linkedIn_logo.svg';
import twitterIcon from '../../../assets/twitter_logo.svg';
import { Container, A, IMG } from './LinkStyled';

export default function Links() {
  return (
    <Container>
      <div>
        <A rel="dns-prefetch" href="https://www.facebook.com/joseph.ireland.773" target="_blank"><IMG src={facebookIcon} alt="facebook icon" /></A>
        <A rel="dns-prefetch" href="https://github.com/IamJoseph" target="_blank"><IMG src={githubIcon} alt="github icon" /></A>
        <A rel="dns-prefetch" href="https://www.linkedin.com/in/joseph-ireland-5832b0a2" target="_blank"><IMG src={linkedInIcon} alt="linkedIn icon" /></A>
        <A rel="dns-prefetch" href="https://twitter.com/JosephIreland6" target="_blank"><IMG src={twitterIcon} alt="twitter icon" /></A>
      </div>
    </Container>
  );
}
