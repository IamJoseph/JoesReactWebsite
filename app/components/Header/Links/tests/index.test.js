import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Links from '../index';
import { IMG, Link, UL } from '../LinksStyled';

describe('<Links />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Links />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls componentDidUpdate when new state added, ul classname is not "hide" if state.open is true', () => {
    const wrapper = shallow(<Links />);
    const spy = jest.spyOn(Links.prototype, 'componentDidUpdate');
    expect(spy).not.toHaveBeenCalled();
    wrapper.setState({ open: true });
    expect(spy).toHaveBeenCalled();
    expect(wrapper.find(UL).hasClass('hide')).toBe(false);
  });

  it('toggles state.open when button pressed', () => {
    const wrapper = shallow(<Links />);
    expect(wrapper.state().open).toEqual(false);
    wrapper.find(IMG).simulate('click');
    expect(wrapper.state().open).toEqual(true);
    wrapper.find(IMG).simulate('click');
    expect(wrapper.state().open).toEqual(false);
  });


  it('renders three links i.e. Home,Work,Contact', () => {
    const wrapper = shallow(<Links />);
    expect(wrapper.find(Link).length).toEqual(3);
  });
});
