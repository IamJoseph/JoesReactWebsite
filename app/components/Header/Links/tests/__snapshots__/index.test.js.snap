// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> renders correctly 1`] = `
<styled.nav>
  <styled.div>
    <styled.div>
      <styled.img
        alt="hamburger menu icon"
        onClick={[Function]}
        src="IMAGE_MOCK"
      />
    </styled.div>
    <styled.ul
      className="hide"
    >
      <Styled(NavLink)
        activeStyle={
          Object {
            "backgroundColor": "#8aa3b6",
          }
        }
        exact={true}
        to="/"
      >
        Home
      </Styled(NavLink)>
      <Styled(NavLink)
        activeStyle={
          Object {
            "backgroundColor": "#8aa3b6",
          }
        }
        to="/work"
      >
        Work
      </Styled(NavLink)>
      <Styled(NavLink)
        activeStyle={
          Object {
            "backgroundColor": "#8aa3b6",
          }
        }
        to="/contact"
      >
        Contact
      </Styled(NavLink)>
    </styled.ul>
  </styled.div>
  <Styled(NavLink)
    activeClassName="active"
    exact={true}
    to="/"
  >
    Home
  </Styled(NavLink)>
  <Styled(NavLink)
    activeClassName="active"
    to="/work"
  >
    Work
  </Styled(NavLink)>
  <Styled(NavLink)
    activeClassName="active"
    to="/contact"
  >
    Contact
  </Styled(NavLink)>
</styled.nav>
`;
